embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/config/source/OkHttpRemoteConfigSource.kt

```diff
@@ -27,29 +28,47 @@
         null
     }
 
+    override fun setInitialEtag(etag: String) {
+        this.etag = etag
+    }
+
+    private var etag: String? = null
+
     private fun fetchConfigImpl(): RemoteConfig? {
+        val request = prepareRequest()
+        val call = okhttpClient.newCall(request)
+        val response = call.execute()
+        return processResponse(response)
+    }
+
+    private fun prepareRequest(): Request {
         val url = apiUrlBuilder.resolveUrl(Endpoint.CONFIG)
         val headers = prepareConfigRequest(url).getHeaders()
         val builder = Request.Builder().url(url)
 
+        etag?.let {
```

Comment:
Just to make sure, will OkHttp actually short circuit the request and return without actually pinging the server if the etag matches what it has in its local cache?

Reply:
Yes - there's test coverage for this use-case and I've also manually tested this out by launching the example app twice with a debugger attached, and confirmed a 304 response came from the live endpoint

Reply:
To further clarify how the etag implementation works:

1. Client makes request to config endpoint
2. Server responds 200 with config in body
3. Client reads etag from response header & saves it for later
4. Client makes another request to the config endpoint, which includes a header with the previous etag
5. Server responds 304 with no config in body
6. Client reads status code & knows that it should rely on whatever response was stored locally